src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -173,6 +173,23 @@ bool ClusterTaskManager::PoppedWorkerHandler(
     not_detached_with_owner_failed = true;
   }
 
+  // Check whether the placement group was removed or not.
```

Comment:
I don't think we need to check placement group here. Just check if resources still exist, and just comment when this can be removed. 

```cpp
// check if the resource still exists. It can happen if placement group is removed (i.e., the custom resources are removed).
    const auto &required_resource =
        task.GetTaskSpecification().GetRequiredResources().GetResourceMap();
    for (auto &entry : required_resource) {
      if (!cluster_resource_scheduler_->ContainResource(entry.first)) {
        RAY_LOG(DEBUG) << "The placement group: "
                       << task.GetTaskSpecification().PlacementGroupBundleId().first
                       << " was removed when poping workers for task: " << task_id
                       << ", will cancel the task.";
        CancelTask(task_id, rpc::RequestWorkerLeaseReply::PLACEMENT_GROUP_REMOVED);
        canceled = true;
      }
    }
```

Reply:
do you mean:
1. Check if the resource that the current task required was removed from the local resource view.
2. If so, check whether the task has a corresponding pg, if so, do `CancelTask(type= PLACEMENT_GROUP_REMOVED)` ?

Reply:
then, What cancel type should we set if the task doesn't specify a pg? are there other scenarios that will encounter the same issue?

Reply:
This shouldn't happen other than pg scenario right? 

Maybe we can do 

```
    const auto &required_resource =
        task.GetTaskSpecification().GetRequiredResources().GetResourceMap();
    for (auto &entry : required_resource) {
      if (!cluster_resource_scheduler_->ContainResource(entry.first)) {
        RAY_CHECK(task.GetTaskSpecification().PlacementGroupBundleId().first !=
      PlacementGroupID::Nil());
        RAY_LOG(DEBUG) << "The placement group: "
                       << task.GetTaskSpecification().PlacementGroupBundleId().first
                       << " was removed when poping workers for task: " << task_id
                       << ", will cancel the task.";
        CancelTask(task_id, rpc::RequestWorkerLeaseReply::PLACEMENT_GROUP_REMOVED);
        canceled = true;
      }
    }
```

Reply:
updated.